bridge/evm/contracts/BridgeLimiter.sol

```diff
@@ -168,9 +192,19 @@ contract BridgeLimiter is IBridgeLimiter, CommitteeUpgradeable, OwnableUpgradeab
             "BridgeLimiter: Source chain not supported"
         );
 
-        // update the limit
-        totalLimit = newLimit;
+        // update the chain limit
+        chainLimits[sourceChainID] = newLimit;
 
         emit LimitUpdated(sourceChainID, newLimit);
     }
+
+    /* ========== PRIVATE FUNCTIONS ========== */
+
+    function getChainHourTimestampKey(uint8 chainID, uint32 hourTimestamp)
+        public
+        pure
+        returns (uint256)
+    {
+        return (uint256(chainID) << 32) | uint256(hourTimestamp);
+    }
```

Comment:
1. is this a common way in solidity to substitute a tuple key in mapping? and how much more efficient is this compared to nested mapping?
2. this is not `/* ========== PRIVATE FUNCTIONS ========== */`

Reply:
1. Nested mappings are not as gas efficient as the composite key approach. Trade off is it's less readable... both approaches are very common. 
2. Good call